library/gui/screen/src/main/java/org/quiltmc/qsl/screen/api/client/ScreenEvents.java

```diff
@@ -0,0 +1,210 @@
+	 * It should be noted some of the methods in {@link QuiltScreenHooks} such as a screen's {@link QuiltScreenHooks#getTextRenderer(Screen) text renderer} may not be initialized yet, and as such their use is discouraged.
+	 *
+	 * <!--<p>Typically this event is used to register screen events such as listening to when child elements are added to the screen. ------ Uncomment when child add/remove event is added for elements-->
+	 * You can still use {@link ScreenEvents#AFTER_INIT} to register events such as keyboard and mouse events.
+	 *
+	 * <p>The {@link ScreenExtensions} provided by the {@code info} parameter may be used to register tick, render events, keyboard, mouse, additional and removal of child elements (including buttons).
+	 * For example, to register an event on inventory like screens after render, the following code could be used:
+	 * <pre>{@code
+	 * &#64;Override
+	 * public void onInitializeClient() {
+	 * 	ScreenEvents.BEFORE_INIT.register((client, screen, scaledWidth, scaledHeight) -> {
+	 * 		if (screen instanceof AbstractInventoryScreen) {
+	 * 			ScreenEvents.getAfterRenderEvent(screen).register((matrices, mouseX, mouseY, tickDelta) -> {
+	 * 				...
+	 * 			});
+	 * 		}
+	 * 	});
+	 * }
+	 * }</pre>
+	 *
+	 * <p>This event indicates a screen has been resized, and therefore is being re-initialized.
+	 * This event can also indicate that the previous screen has been changed.
+	 * @see ScreenEvents#AFTER_INIT
+	 */
+	public static final Event<BeforeInit> BEFORE_INIT = Event.create(BeforeInit.class, callbacks -> (client, screen, scaledWidth, scaledHeight) -> {
+		for (BeforeInit callback : callbacks) {
+			callback.beforeInit(client, screen, scaledWidth, scaledHeight);
+		}
+	});
+
+	/**
+	 * An event that is called after {@link Screen#init(MinecraftClient, int, int) a screen is initialized} to it's default state.
+	 *
+	 * <p>Typically this event is used to modify a screen after the screen has been initialized.
+	 * Modifications such as changing sizes of buttons, removing buttons and adding/removing child elements to the screen can be done safely using this event.
+	 *
+	 * <p>For example, to add a button to the title screen, the following code could be used:
+	 * <pre>{@code
+	 * ScreenEvents.AFTER_INIT.register((client, screen, scaledWidth, scaledHeight) -> {
+	 * 	if (screen instanceof TitleScreen) {
+	 * 		Screens.getButtons(screen).add(new ButtonWidget(...));
+	 * 	}
+	 * });
+	 * }</pre>
+	 *
+	 * <p>Note that by adding an element to a screen, the element is not automatically {@link net.minecraft.client.gui.screen.Screen ticked} or {@link net.minecraft.client.gui.Drawable drawn}.
+	 * Unless the element is button, you need to call the specific {@link Screen#tick() tick} and {@link net.minecraft.client.gui.Drawable#render(MatrixStack, int, int, float) render} methods in the corresponding screen events.
+	 *
+	 * <p>This event can also indicate that the previous screen has been closed.
```

Comment:
This should be with the first few sentences explaining the event.

Reply:
Fixed.